modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/SW360MetaDataUpdater.java

```diff
@@ -55,22 +55,21 @@ public SW360MetaDataUpdater(SW360ConnectionConfiguration sw360ConnectionConfigur
         this.uploadSources = uploadSources;
     }
 
-    public Set<String> getOrCreateLicenses(Artifact artifact) {
+    public Map<License, Optional<SW360License>> getLicenses(Artifact artifact) {
```

Comment:
Keys of the map are never accessed, why not instead use `List<Optional<SW360License>>`. And since, if at least one entry is empty, it is treated as empty. It might be even better to use `Optional<List<SW360License>>`.

The function `licenseClientAdapter.getSW360LicenseByAntennaLicense(license, header)` has no side effect? Why is the early exit solution gone?

Reply:
I do not agree, you are right, that the keys are not used, but I do not like lists with occacional empty stuff that lost its meaning, by using the Map, an empty value has meaning. In addition, I think this method might be usabe in other context, too. In the current change, this might already be the function I need.

I do not get your remark on the early exit solution? You mean that the method returns an empty list if at least one license is not available. I didn't like this because the method in this place should not be aware that if one license is missing, no information should be given back. This is not intuitive. So that was the reason, why I reimplemented this to return all known license and to let the updater decide what to do in case of missing licenses.

Reply:
I thought you prefer simpler and shorter code, which does not try to solve issues which might appear in the future.

Then I would say that `Map<License, Optional<SW360License>>` is a very complicated datastructure, and it might be good to introduce a custom class to explain the implicit semantics. Or maybe add a comment.

Reply:
Have a look at my new change, I reversed the map thing, but changed the parameter slightly. I even think the functionality is now better distributed. And it allows to compare the numbers at the right place.

Reply:
Ups sorry, I answered in the wrong place, please have a look at my latest change, it should be more appropriate